tests/components/shelly/test_sensor.py

```diff
@@ -358,6 +358,23 @@ async def test_block_sensor_unknown_value(
     assert hass.states.get(entity_id).state == STATE_UNKNOWN
 
 
+async def test_block_sensor_unknown_option_value(
+    hass: HomeAssistant, mock_block_device: Mock, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Test block sensor unknown option value."""
+    entity_id = f"{SENSOR_DOMAIN}.test_name_operation"
+    await init_integration(hass, 1)
+
+    assert hass.states.get(entity_id).state == "normal"
+
+    monkeypatch.setattr(
+        mock_block_device.blocks[SENSOR_BLOCK_ID], "sensorOp", "unknown"
+    )
+    mock_block_device.mock_update()
+
+    assert hass.states.get(entity_id).state is None
```

Comment:
```suggestion
    assert hass.states.get(entity_id).state == STATE_UNKNOWN
```

Reply:
The PR is a consequence of @bieniu requesting to remove "unknown" from the state of the sensors.
So I adapted the code to report `None` instead of unknown and was trying to make a test for this.

Reply:
Simone, take a look here:

https://github.com/home-assistant/core/blob/940625505f56f0adef0b0978927d5072ab139250/homeassistant/helpers/entity.py#L1029-L1030

Reply:
So we are back to the beginning: if the state is  `None` or `unknown` it always return `unknown`.

No need to remove it from the state list then.

Reply:
No, our code should return `None`, not `unknown`.

Reply:
> The PR is a consequence of @bieniu requesting to remove "unknown" from the state of the sensors. So I adapted the code to report `None` instead of unknown and was trying to make a test for this.

I commented about the **test**, not about removing `unknown`, the test failed because it expects `None` but `None` is interpreted as `unknown` **which actually shows that we should remove this state and let core handle this correctly.**